martha/martha_v3.js

```diff
@@ -0,0 +1,68 @@
+const {dataObjectUriToHttps, parseRequest} = require('../common/helpers');
+const {bondBaseUrl, determineBondProvider, BondProviders} = require('../common/bond');
+const apiAdapter = require('../common/api_adapter');
+
+
+async function maybeTalkToBond(req, provider = BondProviders.default) {
+    // Currently HCA data access does not require additional credentials.
+    // The HCA checkout buckets allow object read access for GROUP_All_Users@firecloud.org.
+    if (req && req.headers && req.headers.authorization && provider !== BondProviders.HCA) {
+        try {
+            return await apiAdapter.getJsonFrom(`${bondBaseUrl()}/api/link/v1/${provider}/serviceaccount/key`, req.headers.authorization);
+        } catch (error) {
+            console.log(`Received error while fetching service account from Bond for provider '${provider}'.`);
+            console.error(error);
+            throw error;
+        }
+    } else {
+        return Promise.resolve();
+    }
+}
+
+function aggregateResponses(responses) {
+    // Note: for backwards compatibility, we are returning the DRS object with the key, "dos".  If we want to change this, we can add a new api version for Martha_v2.
+    // When/if we change this, we might want to consider replacing "dos" with "data_object" or something like that that is unlikely to change
+    const finalResult = { dos: responses[0] };
```

Comment:
Yeah, one step at time... can't wait to fix up the replies. Right now `martha_v3` returns very different `response[0]` objects for JDR compared to the other repos.

Reply:
Yes. Currently whatever JDR returns it will be the value for `dos` key. It will change in WA-160 once there is a standard response schema.